sql/moz-fx-data-shared-prod/telemetry_derived/sponsored_tiles_clients_daily_v1/query.sql

```diff
@@ -0,0 +1,299 @@
+------ DESKTOP SPONSORED TILES
+WITH newtab_unnested AS (
+  SELECT AS STRUCT
+    t.client_info.client_id,
+    date(t.submission_timestamp) AS submission_date,
+    t.sample_id,
+    s.name,
+    s.category,
+    s.extra
+  FROM
+    `mozdata.firefox_desktop.newtab` t
+  CROSS JOIN
+    UNNEST(t.events) s
+  WHERE
+    date(t.submission_timestamp) = @submission_date
+),
+desktop_events_1 AS (
+  -- desktop tiles clicks and impressions
+  SELECT
+    client_id,
+    submission_date,
+    COUNTIF(
+      name = "click"
+      AND category = "topsites"
+      AND mozfun.map.get_key(extra, "is_sponsored") = "true"
+    ) AS sponsored_tiles_click_count,
+    COUNTIF(
+      name = "impression"
+      AND category = "topsites"
+      AND mozfun.map.get_key(extra, "is_sponsored") = "true"
+    ) AS sponsored_tiles_impression_count,
+  FROM
+    newtab_unnested
+  WHERE
+    submission_date = @submission_date
+  GROUP BY
+    1,
+    2
+),
+desktop_events_2 AS (
+-- desktop Sponsored Tile Dismissals and Disables
+  SELECT
+    client_id,
+    DATE(submission_timestamp) AS submission_date,
+    COUNTIF(
+      event = 'BLOCK'
+      AND value LIKE '%spoc%'
+      AND value LIKE '%card_type%'
+    ) AS sponsored_tiles_dismissal_count,
+    COUNTIF(
+      event = 'PREF_CHANGED'
+      AND source = 'SPONSORED_TOP_SITES'
+    ) AS sponsored_tiles_disable_count
+  FROM
+    `mozdata.activity_stream.events`
+  WHERE
+    DATE(submission_timestamp) = @submission_date
+  GROUP BY
+    1,
+    2
+),
+desktop_clients AS (
+  SELECT
+    os,
+    submission_date,
+    client_id,
+    browser_version_info,
+    experiments,
+    country,
+    locale,
+    normalized_channel,
+    normalized_os_version,
+    profile_age_in_days,
+    sample_id
+  FROM
+    `moz-fx-data-shared-prod.telemetry.clients_daily`
+  WHERE
+    submission_date = @submission_date
+  -- Desktop Sponsored Tiles is only available for the following clients:
+    AND country IN UNNEST(["AU", "BR", "CA", "DE", "ES", "FR", "GB", "IN", "IT", "JP", "MX", "US"])
+    AND browser_version_info.major_version >= 92
+    AND browser_version_info.version NOT IN ('92', '92.', '92.0', '92.0.0')
+),
+------ iOS SPONSORED TILES METRICS
+ios_events AS (
+  -- iOS clicks and impressions
+  SELECT
+    client_info.client_id,
```

Comment:
Just wanted to make sure, does `mozdata.firefox_ios.events_unnested` have the `client_id` field? I am not super familiar with this data set, but AFAIK, the topsites data collections (i.e. where we store impressions & clicks) for mobile use `context_id` rather than `client_id`.

Reply:
Yes, iOS events_unnested has client_id as it's not one of the contextual_services tables